### PR TITLE
[Fix #8281] Improve WhileUntilModifier detection and correction

### LINKED ISSUE #8281: Style/WhileUntilModifier must copy logic from Style/IfUnlessModifier
There's a lot of logic in `IfUnlessModifier` that is missing from `WhileUntilModifier`.

1) Looses comments:

```ruby
while bar # important comment about to be nuked
  baz
end
# corrected to
baz while bar
```

2) Incorrect correction for assignment:

```ruby
foo = while bar
  baz
end
# corrected to
foo = baz while bar
```

This is logic that `IfUnlessModifier` has.

It should be moved into their shared module, and the corresponding tests should be moved in their shared context.

3) Doesn't detect inline cases:

```ruby
while foo; bar; end # => no offense
```

Again, `IfUnless` does the right thing. Same logic should be applied, tests should be moved in their shared context.

Found while checking #8273. See also #8282 and #8283

### PR DESCRIPTION
**Pull request by Darhazer**
The extracted code was already there, it just had to be re-used. Moved relevant tests to the shared examples.

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop-hq/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by marcandre** [comment]
Looks good.
Is the moving of many specs directly related to this change?

**Comment by marcandre** [comment]
(and wouldn't the `if` form also benefit from the same?)

File: changelog/fix_while_until_modifier_autocorrection.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* [#8281](https://github.com/rubocop-hq/rubocop/issues/8281) Fix Style/WhileUntilModifier handling comments and assignment when correcting to modifier form. ([@Darhazer][])
```
Comment (bbatsov) [inline changelog/fix_while_until_modifier_autocorrection.md:1]: I think you forgot `:` here.

Comment (Darhazer) [reply changelog/fix_while_until_modifier_autocorrection.md]: Sorry about that :blush: Fixed

**Comment by Darhazer** [comment]
> Is the moving of many specs directly related to this change?

The change makes those pass. You mentioned in the original ticket that "the corresponding tests should be moved in their shared context", and I agree - the actual code is also in the shared module. Copy/pasting those specs doesn't make sense.

> (and wouldn't the if form also benefit from the same?)

What do you mean?

**Comment by marcandre** [comment]
> > Is the moving of many specs directly related to this change?
> 
> The change makes those pass.

Oh, perfect then. I thought they might have already been passing

> > (and wouldn't the if form also benefit from the same?)
> 
> What do you mean?

There's a spec being added to `while_until_modifier_spec`. Shouldn't that spec be added to `condition_modifier_cop.rb ` instead (and made to pass)?
I.e.
```ruby
# bad
if foo; bar; end
# good
bar if foo
```

**Comment by Darhazer** [comment]
> There's a spec being added to while_until_modifier_spec. Shouldn't that spec be added to condition_modifier_cop.rb instead (and made to pass)?

Sure. Done. Thank you :bow:

**Review by marcandre — APPROVED** [review]
LGTM 👍

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**